wire-ios-data-model/Source/ConversationSearch/SearchableConversation.swift

```diff
@@ -0,0 +1,25 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+/// A protocol which represents a conversation that can be searched for.
+public protocol SearchableConversation {
+    associatedtype SearchableParticipant: SearchableConversationParticipant
+
+    var searchableName: String { get }
+    var searchableParticipants: [SearchableParticipant] { get }
+}
```

Comment:
Is the purpose to conceal managed objects from the view layer?

Reply:
Yes, to make the search algorithm independent from the CoreData entities.